controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -399,6 +399,7 @@ func (w *Workload) UpdateKubeProxyImageInfo(ctx context.Context, kcp *controlpla
 	}
 
 	if container.Image != newImageName {
+		// NOTE: This object doesn't implement observedGeneration. No need to add the WithStatusObservedGeneration option.
```

Comment:
This comment seems a bit misleading, we shouldn't be updating observedGeneration because we are not managing this resource, just trying to get information from an externally reconciled resource.